onchain/src/art_peace.cairo

```diff
@@ -982,7 +1004,31 @@ pub mod ArtPeace {
             // TODO: Allow some threshold?
             if matches == template_metadata.width * template_metadata.height {
                 self.templates.completed_templates.write(template_id, true);
-            // TODO: Distribute rewards
+                // Distribute rewards
+                if pixel_contributors.len() > 0 {
+                    let mut i = 0;
+
+                    while i < pixel_contributors
+                        .len() {
+                            let reward_token = template_metadata.reward_token;
+                            let reward_amount = template_metadata.reward;
+                            let total_pixels = self.total_pixels.read();
+
+                            let mut user = *pixel_contributors.at(i).into();
+                            let user_index = (i + 1);
+                            let user_total_pixels = total_pixels_by_user.get(user_index.into());
+                            
+                            let user_reward = (reward_amount * user_total_pixels.into()) / total_pixels.into();
```

Comment:
`total_pixels` is the number of pixels in canvas.
You have to divide with number of pixels in template, i.e.: `template_metadata.width * template_metadata.height`